src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -734,6 +733,21 @@ public static string FormatPackageVersionList(IEnumerable<Tuple<PackageHeader, P
         /// <param name="packageDownloadHandle">package download handle</param>
         internal virtual Task<(PackageDownloadHandle handle, string downloadPath)> Download(PackageDownloadHandle packageDownloadHandle)
         {
+            // We only want to display the last 3 downloaded packages to the user
+            // in the form of toast notifications.
+            if (Downloads.Count >= 3)
+            {
+                // Iterates backwards through the downloads list and removes items
```

Comment:
you should be able to use `Downloads.RemoveRange(index: 2, count: Downloads.Count - 2);` instead of this whole for loop

Reply:
Have just tested and this doesn't quite do what we want - this only replaces the last item if there's more than 3 in the list. The UI here displays the list like a stack, so the most recent downloads appear at the bottom, above that the 2nd most recent, above that the 3rd etc. The oldest item on the stack is 'popped' off when a new download happens.

Reply:
Sorry for the slightly long GIF, but this is what happens:

![xLrqrxNX9S](https://user-images.githubusercontent.com/29973601/134888494-915155f9-78fe-46e1-866e-a3b207085d2a.gif)

Reply:
Oh, I misunderstood how that list in interpreted by the UI
Then you should be able to use `Downloads.RemoveRange(index: 0, count: Downloads.Count - 2);`
That should basically remove from the beginning of the list (i.e index 0) until only 2 elements remain in the list

Reply:
Works great, thanks :)